nomenclature/config.py

```diff
@@ -101,24 +98,60 @@ def check_repository_consistency(cls, values):
                 )
         return values
 
+    @property
+    def repos(self) -> Dict[str, str]:
+        return {
+            dimension: getattr(self, dimension).repository
+            for dimension in ("region", "variable")
+            if getattr(self, dimension) and getattr(self, dimension).repository
+        }
+
+
+class RegionMappingConfig(BaseModel):
+    repository: str
+
+
+class NomenclatureConfig(BaseModel):
+    repositories: Dict[str, Repository] = {}
+    definitions: Optional[DataStructureConfig]
+    mappings: Optional[RegionMappingConfig]
+
+    @root_validator
+    def check_definitions_repository(cls, v):
+        definitions_repos = v.get("definitions").repos if v.get("definitions") else {}
+        mapping_repos = (
+            {"mappings": v.get("mappings").repository} if v.get("mappings") else {}
+        )
+        repos = {**definitions_repos, **mapping_repos}
+        if repos and not v.get("repositories"):
```

Comment:
I guess this error is overly pydantic - the next check will catch it if the repository is not defined, right?

Reply:
It would produce a different kind of error. 
If `repositories` is not defined the check below would resolve as:
```python
if repository not in None
```
which would result in this error:
```console
Traceback (most recent call last):
TypeError: argument of type 'NoneType' is not iterable
```

Reply:
Ok, got it!